src/sources/dnstap/dns_message_parser.rs

```diff
@@ -0,0 +1,2000 @@
+        52 => Some(String::from("TLSA")),
+        53 => Some(String::from("SMIMEA")),
+        55 => Some(String::from("HIP")),
+        56 => Some(String::from("NINFO")),
+        57 => Some(String::from("RKEY")),
+        58 => Some(String::from("TALINK")),
+        59 => Some(String::from("CDS")),
+        60 => Some(String::from("CDNSKEY")),
+        61 => Some(String::from("OPENPGPKEY")),
+        62 => Some(String::from("CSYNC")),
+        63 => Some(String::from("ZONEMD")),
+        99 => Some(String::from("SPF")),
+        100 => Some(String::from("UINFO")),
+        101 => Some(String::from("UID")),
+        102 => Some(String::from("GID")),
+        103 => Some(String::from("UNSPEC")),
+        104 => Some(String::from("NID")),
+        105 => Some(String::from("L32")),
+        106 => Some(String::from("L64")),
+        107 => Some(String::from("LP")),
+        108 => Some(String::from("EUI48")),
+        109 => Some(String::from("EUI64")),
+        249 => Some(String::from("TKEY")),
+        250 => Some(String::from("TSIG")),
+        251 => Some(String::from("IXFR")),
+        252 => Some(String::from("AXFR")),
+        253 => Some(String::from("MAILB")),
+        254 => Some(String::from("MAILA")),
+        255 => Some(String::from("ANY")),
+        256 => Some(String::from("URI")),
+        257 => Some(String::from("CAA")),
+        258 => Some(String::from("AVC")),
+        259 => Some(String::from("DOA")),
+        260 => Some(String::from("AMTRELAY")),
+        32768 => Some(String::from("TA")),
+        32769 => Some(String::from("DLV")),
+
+        _ => None,
+    }
+}
+
+fn format_bytes_as_hex_string(bytes: &[u8]) -> String {
+    bytes
+        .iter()
+        .map(|e| format!("{:02X}", e))
+        .collect::<Vec<String>>()
+        .join(".")
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::{
+        net::{Ipv4Addr, Ipv6Addr},
+        str::FromStr,
+        time::Instant,
+    };
+    use trust_dns_proto::{
+        rr::{
+            dnssec::{
+                rdata::{
+                    dnskey::DNSKEY, ds::DS, nsec::NSEC, nsec3::NSEC3, nsec3param::NSEC3PARAM,
+                    sig::SIG, DNSSECRData,
+                },
+                Algorithm as DNSSEC_Algorithm, DigestType, Nsec3HashAlgorithm,
+            },
+            domain::Name,
+            rdata::{
+                caa::KeyValue,
+                sshfp::{Algorithm, FingerprintType},
+                tlsa::{CertUsage, Matching, Selector},
+            },
+            rdata::{null, CAA, NAPTR, SSHFP, TLSA, TXT},
+        },
+        serialize::binary::Restrict,
+    };
+
+    impl DnsMessageParser {
+        pub fn raw_message_clone(&self) -> Option<&Vec<u8>> {
+            self.raw_message_clone.as_ref()
+        }
+    }
+
+    #[test]
+    fn test_parse_as_query_message() {
+        let raw_dns_message = "szgAAAABAAAAAAAAAmg1B2V4YW1wbGUDY29tAAAGAAE=";
+        if let Ok(raw_qeury_message) = base64::decode(raw_dns_message) {
```

Comment:
Nit: I think you could remove some of the conditional nesting here using `expect` for assertions. For example:

```
let raw_query_message = base64::decode(raw_dns_message).expect("invalid base64 data");
```

Then you don't need to do `if/else`.

This suggestion applies to the below tests too.

Reply:
Thanks Jesse. Will be using `expect` for assertions all the related tests.

Reply:
Changed.